fusion-api/src/main/java/io/yupiik/fusion/framework/api/main/Launcher.java

```diff
@@ -49,16 +54,34 @@ public void await() {
 
     @Override
     public void close() {
-        container.close();
+        if (!isClosing.compareAndSet(false, true)) {
+            container.close();
+        }
     }
 
     protected Instance<List<Awaiter>> lookupAwaiters() {
         return container.lookups(Awaiter.class, list -> list.stream().map(Instance::instance).toList());
     }
 
     public static void main(final String... args) {
+        Thread hook = null;
         try (final var launcher = new Launcher(args)) {
+            Configuration configuration = launcher.container.lookup(Configuration.class).instance();
+            useHook.set(Boolean.parseBoolean(configuration.get("fusion.launcher.useHook").orElse("false")));
```

Comment:
No need of a field instance, can be a local final boolean

Reply:
Agreed!